### PR TITLE
refactor: create common functions

### LINKED ISSUE #3: refactor: create common functions
we should keep the scope of common functions for the plugins here.
ex:

* `splitPluginArguments()`:
https://github.com/malikovmm/scan2html/blob/html-report/main.go#L103-L121
also we should improve this function

* `makeTrivyJsonReport()`:
https://github.com/afdesk/scan2html/blob/main/main.go#L85-L95

* `getPathToTemplate()` and `readPluginFile()` have almost the same logic
https://github.com/malikovmm/scan2html/blob/html-report/main.go#L123-L129
https://github.com/afdesk/csv-report/blob/main/main.go#L121-L127

### PR DESCRIPTION
**Pull request by nikpivkin**
Closes #3

### CODE REVIEW

File: command/command.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+type Arguments map[string]string
+
+func RetrievePluginArguments(availableArguments []string) (pluginArgs Arguments, rest []string) {
+	trivyCommand := make([]string, 0, len(os.Args))
+	args := make(map[string]string)
+	for i := 0; i < len(os.Args); i++ {
+		if slices.Contains(availableArguments, os.Args[i]) {
+			args[os.Args[i]] = os.Args[i+1]
+			i++ // skip argument value
```
Comment (afdesk) [inline command/command.go:64]: if an availableArgument is the last there will be `panic` here.
let's use an empty string for this case

**Review by afdesk — CHANGES_REQUESTED** [review]

File: command/command.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	path, err := GetPathToPluginDir(fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	return os.ReadFile(path)
+}
+
+func MakeTrivyJsonReport(trivyCommand []string, outputFileName string) error {
+	cmdArgs := append(trivyCommand, "--format", "json", "--output", outputFileName)
```
[the remark this answers is not in the document]
Comment (nikpivkin) [reply command/command.go:46]: @afdesk https://github.com/afdesk/scan2html/blob/8c9616cc68b1a9a3d25cc63dd8013c7ad5058ec5/main.go#L87
https://github.com/nikpivkin/csv-report/blob/0c17278adc06ff50c0e29a8e1a4c9ab1af8bf415/main.go#L55

[the remark this answers is not in the document]
Comment (afdesk) [reply command/command.go:46]: i saw these lines, so I wrote that couldn't find the solution.
please, added a check for it too here

[the remark this answers is not in the document]
Comment (nikpivkin) [reply command/command.go]: @afdesk we can skip this, it will be Trivy to use our arguments, because they come later

**Review by afdesk — APPROVED** [review]

**Pull request merged**